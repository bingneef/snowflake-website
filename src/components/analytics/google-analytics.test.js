import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { render, fireEvent, cleanup } from "react-testing-library";

import GoogleAnalytics, { sendEvent } from "./google-analytics";

describe("GoogleAnalytics", () => {
  describe("env = development", () => {
    beforeEach(() => {
      process.env.NODE_ENV = "development";
    });

    it("renders without error", () => {
      const div = document.createElement("div");

      ReactDOM.render(<GoogleAnalytics />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it("matches the snapshot", () => {
      const component = renderer.create(<GoogleAnalytics />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("env = production", () => {
    beforeEach(() => {
      process.env.NODE_ENV = "production";
    });

    it("renders without error", () => {
      const div = document.createElement("div");

      ReactDOM.render(<GoogleAnalytics />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it("matches the snapshot", () => {
      const component = renderer.create(<GoogleAnalytics />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

describe("#sendEvent", () => {
  it("logs if ga is not defined", () => {
    console.log = jest.fn();
    window.ga = undefined;

    sendEvent({ eventCategory: "TestEvent", eventAction: "Sending Event" });

    expect(console.log).toHaveBeenCalledWith(
      "Not sending event: TestEvent:Sending Event, not in production environment"
    );
  });

  it("sends to ga if defined", () => {
    window.ga = {
      send: jest.fn()
    };

    sendEvent({
      eventCategory: "TestEvent",
      eventAction: "Sending Event",
      eventLabel: "Event label",
      eventValue: "EventValue"
    });

    expect(window.ga.send).toHaveBeenCalledWith({
      hitType: "event",
      eventCategory: "TestEvent",
      eventAction: "Sending Event",
      eventLabel: "Event label",
      eventValue: "EventValue"
    });
  });
});
