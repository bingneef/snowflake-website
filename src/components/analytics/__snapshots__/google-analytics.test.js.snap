// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleAnalytics env = development matches the snapshot 1`] = `null`;

exports[`GoogleAnalytics env = production matches the snapshot 1`] = `
Array [
  <script
    async={true}
    src="https://www.googletagmanager.com/gtag/js?id=UA-77301580-1"
  />,
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());

            gtag('config', 'UA-77301580-1', { 'optimize_id': 'GTM-TD486C3'});
          ",
      }
    }
  />,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".async-hide { opacity: 0 !important} ",
      }
    }
  />,
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "(function(a,s,y,n,c,h,i,d,e){s.className+=' '+y;h.start=1*new Date;
h.end=i=function(){s.className=s.className.replace(RegExp(' ?'+y),'')};
(a[n]=a[n]||[]).hide=h;setTimeout(function(){i();h.end=null},c);h.timeout=c;
})(window,document.documentElement,'async-hide','dataLayer',4000,
{'OPT_CONTAINER_ID':true});",
      }
    }
  />,
]
`;
